internal/controller/resourcesetinputprovider_controller.go

```diff
@@ -169,13 +186,20 @@ func (r *ResourceSetInputProviderReconciler) reconcile(ctx context.Context,
 func (r *ResourceSetInputProviderReconciler) newGitProvider(ctx context.Context,
 	obj *fluxcdv1.ResourceSetInputProvider,
 	certPool *x509.CertPool,
-	username, password string) (gitprovider.Interface, error) {
+	username, password, ghAppToken string) (gitprovider.Interface, error) {
 	switch {
 	case strings.HasPrefix(obj.Spec.Type, "GitHub"):
+		if password != "" && ghAppToken != "" {
```

Comment:
Let's call `getGitHubAppToken` in here, instead of the reconcile function, no point in calling that function unless we're connecting to GitHub.

Reply:
fixed, thanks!